build/common/installer/scripts/tomlparser-geneva-config.rb

```diff
@@ -0,0 +1,130 @@
+      return nil
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError("Exception while parsing config map for agent settings : #{errorStr}, using defaults, please check config map for errors")
+    return nil
+  end
+end
+
+# Use the ruby structure created after config parsing to set the right values to be used as environment variables
+def populateSettingValuesFromConfigMap(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      
+      geneva_logs_config = parsedConfig[:agent_settings][:geneva_logs_config]
+      if !geneva_logs_config.nil?
+        puts "config: parsing geneva_logs_config settings"
+        genevaEnvironment = geneva_logs_config[:environment]
+        genevaAccount = geneva_logs_config[:account]
+        genevaNamespace = geneva_logs_config[:namespace]
+        genevaConfigVersion = geneva_logs_config[:configversion]
+        if !genevaEnvironment.nil? && !genevaAccount.nil? && !genevaNamespace.nil? && !genevaConfigVersion.nil?
+          if GENEVA_SUPPORTED_ENVIRONMENTS.include?(genevaEnvironment)
+            @genevaEnvironment = genevaEnvironment
+            @genevaAccount = genevaAccount
+            @genevaNamespace = genevaNamespace
+            @genevaConfigVersion = genevaConfigVersion
+          else
+            puts "config::error:unsupported geneva config environment"
+          end
+        else
+          puts "config::error:invalid geneva logs config"
+        end
+        genevaAuthId = geneva_logs_config[:authid]
+        if !genevaAuthId.nil?
+          if genevaAuthId.start_with?("client_id#") || genevaAuthId.start_with?("object_id#") || genevaAuthId.start_with?("mi_res_id#")
+            @genevaAuthId = genevaAuthId
+          else
+            puts "config:error: auth id must be in one of the suppported formats: object_id#<guid> or client_id#<guid> or mi_res_id#<identity resource id>"
+          end
+        end
+        puts "config::info:successfully parsed geneva_logs_config settings"
+      end
+    end
+  rescue => errorStr
+    puts "config::error:Exception while reading config settings for agent configuration setting - #{errorStr}, using defaults"
+  end
+end
+
+# Write the settings to file, so that they can be set as environment variables
+def writeEnvScript(filepath)
+  file = File.open(filepath, "w")
+
+  if !file.nil?
+
+    if !@genevaEnvironment.empty? && !@genevaAccount.empty? && !@genevaNamespace.empty? && !@genevaConfigVersion.empty? && !@genevaAuthId.empty?
+      file.write(get_command_windows("MONITORING_GCS_ENVIRONMENT", @genevaEnvironment))
+      file.write(get_command_windows("MONITORING_GCS_ACCOUNT", @genevaAccount))
+      file.write(get_command_windows("MONITORING_GCS_NAMESPACE", @genevaNamespace))
+      file.write(get_command_windows("MONITORING_CONFIG_VERSION", @genevaConfigVersion))
+
+      authIdParts =  @genevaAuthId.split('#', 2)
```

Comment:
do we need these and I see these are different from Linux? in case of Linux AMA, I am using "MONITORING_GCS_AUTH_ID".

Reply:
Fixed